extended/src/main/java/io/kubernetes/client/extended/kubectl/KubectlRollout.java

```diff
@@ -0,0 +1,113 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package io.kubernetes.client.extended.kubectl;
+
+import io.kubernetes.client.common.KubernetesObject;
+import io.kubernetes.client.extended.kubectl.exception.KubectlException;
+import io.kubernetes.client.extended.kubectl.rollout.DaemonSetRollbacker;
+import io.kubernetes.client.extended.kubectl.rollout.DeploymentRollbacker;
+import io.kubernetes.client.extended.kubectl.rollout.Rollbacker;
+import io.kubernetes.client.extended.kubectl.rollout.StatefulSetRollbacker;
+import io.kubernetes.client.extended.kubectl.rollout.response.RolloutResponse;
+import io.kubernetes.client.openapi.models.V1DaemonSet;
+import io.kubernetes.client.openapi.models.V1Deployment;
+import io.kubernetes.client.openapi.models.V1StatefulSet;
+
+public class KubectlRollout<ApiType extends KubernetesObject>
+    extends Kubectl.ResourceBuilder<ApiType, KubectlRollout<ApiType>>
+    implements Kubectl.Executable<RolloutResponse> {
+
+  private RolloutActionType actionType;
+
+  KubectlRollout(Class<ApiType> apiTypeClass) {
+    super(apiTypeClass);
+  }
+
+  public KubectlRollout<ApiType> history() {
+    this.actionType = RolloutActionType.HISTORY;
```

Comment:
Perhaps we want to throw if someone does multiple types? (e.g. if you call `history().resume()` the code should probably throw)